zerver/lib/message.py

```diff
@@ -229,6 +236,19 @@ def messages_for_ids(
 
     for message_id in message_ids:
         msg_dict = message_dicts[message_id]
+        if user_profile and msg_dict["recipient_type"] == Recipient.STREAM:
+            stream_id = msg_dict["recipient_type_id"]
+            stream = get_stream_by_id_in_realm(stream_id, realm)
+            if (
+                stream.is_support_stream()
+                and not check_access_based_on_can_access_stream_topics_group(user_profile, stream)
+            ):
+                topic_creator_user_id = get_topic_creator_user_id(
```

Comment:
Similarly, doing these queries in a loop is also a problem. Is there a way where we can just filter out the inaccessible messages initially when fetching message IDs?

Reply:
Here are [the](https://chat.zulip.org/#narrow/stream/101-design/topic/Support.20Streams/near/1943912) problems I faced when filtering out the inaccessible messages initially when fetching message IDs